sensors/nxt_i2c_sensor_defs.c

```diff
@@ -25,6 +25,36 @@
 #include "ms_ev3_smux.h"
 #include "ms_nxtmmx.h"
 
+/*
+ * convert raw data of LEGO NXT Temperature Sensor to Celsius
+ */
+static int lego_nxt_temp_celsius_scale(void *context,
```

Comment:
> Great, I was hoping that was the case. What I was concerned about was the fact that min_raw is u32. Does this get cast to u16 on the basis of data_type?

No, this is something we need to look into more. We might actually need to make raw min/max s64.

> Reading the scaling data on the ev3sources typedata page, I'm confused since it states raw min is -14080.0 and raw max is 32767.0. These don't match the data sheet which give 0xc90 (-880) and 0x7ff (2047) respectively.

We are using 12-bit data mode, so the last 4 bits are always 0. We aren't shifting the bits >> 4, so our raw values will actually be -880 * 16 = -14080 and 2047 * 16 = 32752 (LEGO must have used 32767 out of habit). 

> I'm also curious how the 262.4°F Si Max is handled given the integral nature of si_max (below I just truncated - should I be working in x10^decimals there?)

We can't do floating point in the kernel, so we have to used fixed decimals, so `si_max` will be 2624 and we will also need to set `decimals = 1` to tell userland that it is x10